distill/core/log.py

```diff
@@ -57,4 +58,37 @@ def to_json(self) -> str:
 
     def to_dict(self) -> JsonDict:
         return self.data.model_dump(by_alias=True)
-
+    
+    def __lt__(self, other):
+        if isinstance(other, Log):
+            return self.id < other.id
+        if isinstance(other, datetime):
+            return self.id.get_datetime() < other.now()
```

Comment:
`other.now()` will get the current datetime, I think instead we should just be able to compare `self.id.get_datetime() < other` if `other` is a datetime object.

Reply:
This assumes `get_datetime()` returns an actual datetime object, but I don't think it does. Tracking down the method starting from the [PKSUID repo](https://github.com/sashahilton00/python-pksuid/blob/master/pksuid/pksuid.py) to the datetime [library docs](https://docs.python.org/3/library/datetime.html#datetime.date.fromtimestamp) suggests it just returns a float.

Indeed, testing it in the terminal shows it just returns POSIX time float. I believe datetime objects have a `timestamp()` method that will return what we want. But we need to be careful to ensure get_datetime() and `other.timestamp()` are operating on the same scale.

Reply:
@EandrewJones looks to me like it does return a datetime object? "`getDatetime() returns a python date object which represents the approximate time (missing microseconds) that the pksuid was created`"

Reply:
I found that Ryan's comparison works, while trying to compare get_datetime() to timestamp() gives an error due to the float return value of timestamp.